lib/apiservers/engine/backends/cache/image_cache.go

```diff
@@ -115,3 +119,51 @@ func (c *ImageCache) GetImages() []*metadata.ImageConfig {
 
 	return result
 }
+
+func (c *ImageCache) GetImageByDigest(digest digest.Digest) *metadata.ImageConfig {
+	c.m.RLock()
+	defer c.m.RUnlock()
+
+	var imageID string
+
+	for _, image := range c.cache {
+		if strings.Contains(image.ImageID, "sha") {
```

Comment:
Should this be `strings.HasPrefix(image.ImageID, "sha")`? At the moment it will both:
a. check the entire string rather than the start
b. match on the string incorrectly if "sha" can be part of the digest (I don't know if it can - is the digest a hex string?).